python-package/xgboost/spark/core.py

```diff
@@ -1027,22 +1049,48 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
                 else:
                     base_margin = None
 
+                data = {}
                 preds = model.predict(
                     X,
                     base_margin=base_margin,
                     validate_features=False,
                     **predict_params,
                 )
-                yield pd.Series(preds)
+                data[pred.prediction] = pd.Series(preds)
+
+                if pred_contrib_col_name:
+                    contribs = pred_contribs(model, X, base_margin)
+                    data[pred.pred_contrib] = pd.Series(list(contribs))
+                    yield pd.DataFrame(data=data)
+                else:
+                    yield data[pred.prediction]
 
         if has_base_margin:
             pred_col = predict_udf(struct(*features_col, base_margin_col))
         else:
             pred_col = predict_udf(struct(*features_col))
 
-        predictionColName = self.getOrDefault(self.predictionCol)
+        prediction_col_name = self.getOrDefault(self.predictionCol)
 
-        return dataset.withColumn(predictionColName, pred_col)
+        if single_pred:
+            dataset = dataset.withColumn(prediction_col_name, pred_col)
+        else:
+            pred_struct_col = "_prediction_struct"
+            dataset = dataset.withColumn(pred_struct_col, pred_col)
+
+            dataset = dataset.withColumn(
+                prediction_col_name, getattr(col(pred_struct_col), pred.prediction)
+            )
+
+            if pred_contrib_col_name:
+                dataset = dataset.withColumn(
+                    pred_contrib_col_name,
+                    array_to_vector(getattr(col(pred_struct_col), pred.pred_contrib)),
```

Comment:
Does this work with cuDF?

Reply:
Let me add some tests for the GPU scenario.

Reply:
It seems not, since the prediction udf has not added cudf support.